File: app/src/main/java/com/chetaru/tribe365_new/API/retrofit/ConstantAPI.java
package com.chetaru.tribe365_new.API.retrofit;

public class ConstantAPI {

    /******************************Login Api ***************************/
    public static final String api_userLogin="userLogin";
    public static final String api_userSetPassword ="userSetPassword";
    public static final String api_forgotPassword="forgotPassword";

    /*********************HPTM APi List *****************************/
    public static final String getLearningCheckList="getLearningCheckList";
    public static final String getPrinciplesList="getPrinciplesList";
    public static final String changeReadStatusOfUserChecklist="changeReadStatusOfUserChecklist";
    public static final String sendTeamFeedbackToSelectedUsers="sendTeamFeedbackToSelectedUsers";
    public static final String getIndividualQuestionnaireList ="getIndividualQuestionnaireList";
    public static final String addIndividualQuestionnaireAnswers="addIndividualQuestionnaireAnswers";
    public static final String teamFeedbackUserAnswerStatus="teamFeedbackUserAnswerStatus";
    public static final String teamFeedbackUserStatus="teamFeedbackUserStatus";
    /*************************************************************************/


    /**************************** Reflaction APi *****************************/
    public static final String postHPTMReflection ="postHPTMReflection";
    public static final String getHptmReflectionDetail="getHptmReflectionDetail";
    public static final String getReflectionChatMessages="getReflectionChatMessages";
    public static final String sendReflectionChatMessage="sendReflectionChatMessage";
    /****************************************************************************/

    /************************ Home Api *****************************/
    public static final String api_addHappyIndex="addHappyIndex";
    public static final String api_getCurrentVersionOfApp="getCurrentVersionOfApp";
    public static final String api_dotDetail="dotDetail";
    public static final String getHomePageDetails ="getHomePageDetails";
    public static final String getDepartmentuserList="getDepartmentUserList";
    public static final String addDOTBubbleRatingsToMultiDepartment ="addDOTBubbleRatingsToMultiDepartment";
    public static final String addHappyIndex ="addHappyIndex";
    public static final String getCurrentVersionOfApp="getCurrentVersionOfApp";
    public static final String api_getBubbleUnReadNotifications="getBubbleUnReadNotifications";
    public static final String getHomePageKudosCount="getHomePageKudosCount";
    public static final String addKudosAward="addKudosAward";
    public static final String addDotValueKudosAward="addDotValueKudosAward";
    public static final String kudoAwardDetail="kudoAwardDetail";
    public static final String dotValuekudoAwardDetail="dotValuekudoAwardDetail";
    /************************************************************************/

    /****************************************** Absent Module *************************/
    //0: present, 1: absent
    public static final String api_userApplyLeave = "userApplyLeave";
    public static final String api_userChangeLeaveStatus = "userChangeLeaveStatus";
    /****************************** HomePage Basic Version ******************/
    public static final String api_getBasicVersionHomeDetails="getBasicVersionHomeDetails";
    /************************************************************************/

    /*************************** DOT Api ************************************/

    public static final String api_addbubbleRatings="addbubbleRatings";
    public static final String api_getBubbleRatingList="getBubbleRatingList";
    public static final String api_getBubbleFromUserRating="getBubbleFromUserRating";
              //not user in app
    public static final String api_getDOTValuesList="getDOTValuesList";
    /************************************************************************/

    /***************************** Action APi *******************************/
    public static final String api_add_rating="ratingsToDotValues";
    public static final String api_actionTierList="actionTierList";
    public static final String getActionDetail="getActionDetail";

    /************************ Know Org ************************************/
    public static final String getAllOfficenDepartments="getAllOfficenDepartments";
    public static final String getDepartmentList="getDepartmentList";
    public static final String getKnowOrganisationDetails ="getKnowOrganisationDetails";
    //for report section
    public static final String getIndividualReport ="getIndividualReport";


    public static final String api_getUserDashboardReport="getUserDashboardReport";
      public static final String api_getHappyIndexMonthCount="getHappyIndexMonthCount";
      public static final String api_getHappyIndexWeeksCount="getHappyIndexWeeksCount";
      public static final String api_getHappyIndexDaysCount="getHappyIndexDaysCount";
    public static String getKudosAward="getKudosAward";
    /****************************************************************************/

    /*************************************** Know Member ************************/
    public static final String getUserByType="getUserByType";
    public static final String getKnowMemberDetails ="getKnowMemberDetails";
    /************************************************************************/

    /***************************** Offloading *******************************/
    public static final String getFeedbackDetail ="getFeedbackDetail";
    public static final String postFeedback="postFeedback";
    public static final String getChatMessages="getChatMessages";
    public static final String iotSendMsg="iotSendMsg";
    public static final String getInboxChatList="getInboxChatList";
    /*********************************************************************/

    /********************************* Notification ********************/
    public static final String getBubbleRatingUnReadNotificationList="getBubbleRatingUnReadNotificationList";
    public static final String getBubbleRatingNotificationList="getBubbleRatingNotificationList";
    public static final String changeNotificationStatus="changeNotificationStatus";
    public static final String readAllNotification="readAllNotification";
    /**********************************************************************/
    /***************************** support **********************************/
    public static final String getSupportHistory="getSupportHistory";
    public static final String addCustomerSupport="addCustomerSupport";
    public static final String getUserChatMessages="getUserChatMessages";
    public static final String sendChatMessage="sendChatMessage";
    /*******************************************************************/

    /********************************* UserProfile **********************/
    public static final String userProfile="userProfile";
    public static final String userInteractionOnApp="userInteractionOnApp";
    public static final String userLogout="userLogout";
    public static final String updateUserProfile="updateUserProfile";
    public static final String updatePasswordWithCurrentPassword="updatePasswordWithCurrentPassword";
    /*******************************************************************************************/


    /********************************* Action **********************************/
    public static final String getActionList="getActionList";
    public static final String deleteAction="deleteAction";
    public static final String updateStatus="updateStatus";
    public static final String getThemeList="getThemeList";
    public static final String updateAction="updateAction";
    public static final String addAction ="addAction";
    public static final String addComment="addComment";
    public static final String listComment="listComment";
    /*********************************************************************************/

    /************************************ Org Studies ******************************/
    public static final String getOrgDashboardReportWithFilter="getOrgDashboardReportWithFilter";
    /*********************************************************************************/

    public static final String freeVersionHomeDetails ="getFreeVersionHomeDetails";


    /*******************************  Questions and Update Api *************************/
    public static final String isCOTanswerDone="isCOTanswerDone";
    public static final String addCOTAnswer="addCOTAnswer"; //team Role Answers
    public static final String api_addCOTfunctionalLensAnswer="addCOTfunctionalLensAnswer"; //personality type Answers
    public static final String api_sot_addSOTanswers="addSOTanswers"; //culture structure Answers
    public static final String api_sot_addSOTQusAns="getSOTquestionAnswers"; //get supper chargaring questions
    public static final String api_sot_updateSOTquestionAnswer="updateSOTquestionAnswer"; // update supper chaeging questions
    public static final String api_addSOTmotivationAnswer="addSOTmotivationAnswer"; //motivatinal Answers
         /*                             Questions Api                 */
    public static final String api_sot_quest="getSotQuestionList"; //culture Structure Questions
    public static final String api_getSOTmotivationUserList="getSOTmotivationUserList";
    public static final String api_getSOTmotivationQuestions="getSOTmotivationQuestions"; // Motivational Questions
    public static final String api_fubction_qusList="getCOTfunctionalLensQuestionsList"; //personality type
    public static final String api_getCOTMapperSummary="getCOTMapperSummary"; //desctiption
    public static final String api_functionallensdetail="getCOTFunctionalLensDetail"; //personality type result
    public static final String getCOTindividualSummary="getCOTindividualSummary";
    public static final String api_get_COTQuestions="getCOTQuestions"; //Team Role Questions
    public static final String api_getCOTfuncLensCompletedAnswers="getCOTfuncLensCompletedAnswers"; //personality type all answers
    public static final String api_getSOTmotivationCompletedAnswer="getSOTmotivationCompletedAnswer"; //Motivational all Answers
            /*                            update Api                  */
    public static final String api_updateCOTteamRoleMapAnswers="updateCOTteamRoleMapAnswers"; //Team role update answers
    public static final String api_updateCOTfunLensAnswers="updateCOTfunLensAnswers";
    public static final String api_updateSOTmotivationAnswers="updateSOTmotivationAnswers"; // motivations Update answers
           /*                         get Graph                      */
    public static final String api_getTribeometerSubGraph="getTribeometerSubGraph";
    public static final String api_getTribeometerMainSubGraph="getTribeometerMainSubGraph";
    /**************************************************************************************/

    /*********************************** Culture Structure ******************************/

    /*************************************************************************************/

    /*************************************

    /*************************************** Personality Type *****************************/

    public static final String api_getPersonalityTypeReport="getPersonalityTypeReport";
    public static final String api_getPersonalityTypeReportSubGraph="getPersonalityTypeReportSubGraph";
    public static final String api_getPersonalityTypeReportUserSubGraph="getPersonalityTypeReportUserSubGraph";
    public static final String api_getPersonalityTypeQuestionList="getPersonalityTypeQuestionList";
    public static final String api_getPersonalityTypeCompletedAnswers="getPersonalityTypeCompletedAnswers";
    public static final String api_updatePersonalityTypeAnswers="updatePersonalityTypeAnswers";
    public static final String api_addPersonalityTypeAnswers="addPersonalityTypeAnswers";
    /***************************************************************************************/

    /*********************************************** Risk Module *****************************/
    public static final String api_riskRegisterList="riskRegisterList";
    public static final String api_riskDetail="riskDetail";
    /**************************************** update Notification push ***********************/
    public static final String api_updatePushNotificationStatus= "updatePushNotificationStatus";



}

File: app/src/main/java/com/chetaru/tribe365_new/KotlinCode/APi/Constant_API.kt
package com.chetaru.tribe365_new.KotlinCode.APi


const val DEVICE_TYPE : String= "1"
const val ROLE : String= "3"

//const val BASE_URL : String = "https://tribe365demo.chetaru.co.uk/api/" //new  production
const val BASE_URL_API: String = "https://console.tribe365.co/api/";  // live URL


/****************************** Login Api  */
const val api_userLogin: String = "userLogin"
const val api_userSetPassword: String = "userSetPassword"
const val api_forgotPassword: String = "forgotPassword"

/********************* HPTM APi List  */
const val getLearningCheckList = "getLearningCheckList"
const val getPrinciplesList = "getPrinciplesList"
const val changeReadStatusOfUserChecklist = "changeReadStatusOfUserChecklist"
const val sendTeamFeedbackToSelectedUsers = "sendTeamFeedbackToSelectedUsers"
const val getIndividualQuestionnaireList = "getIndividualQuestionnaireList"
const val addIndividualQuestionnaireAnswers = "addIndividualQuestionnaireAnswers"
const val teamFeedbackUserAnswerStatus = "teamFeedbackUserAnswerStatus"
const val teamFeedbackUserStatus = "teamFeedbackUserStatus"
/*************************************************************************/


/**************************** Reflaction APi  */
const val postHPTMReflection = "postHPTMReflection"
const val getHptmReflectionDetail = "getHptmReflectionDetail"
const val getReflectionChatMessages = "getReflectionChatMessages"
const val sendReflectionChatMessage = "sendReflectionChatMessage"
/****************************************************************************/


/************************ Home Api  */
const val api_addHappyIndex = "addHappyIndex"
const val api_getCurrentVersionOfApp = "getCurrentVersionOfApp"
const val api_dotDetail = "dotDetail"
const val getHomePageDetails = "getHomePageDetails"
const val getDepartmentuserList = "getDepartmentUserList"
const val addDOTBubbleRatingsToMultiDepartment = "addDOTBubbleRatingsToMultiDepartment"
const val addHappyIndex = "addHappyIndex"
const val getCurrentVersionOfApp = "getCurrentVersionOfApp"
const val api_getBubbleUnReadNotifications = "getBubbleUnReadNotifications"
const val getHomePageKudosCount = "getHomePageKudosCount"
const val addKudosAward = "addKudosAward"
const val addDotValueKudosAward = "addDotValueKudosAward"
const val kudoAwardDetail = "kudoAwardDetail"
const val dotValuekudoAwardDetail = "dotValuekudoAwardDetail"
/************************************************************************/


/****************************************** Absent Module  */ //0: present, 1: absent
const val api_userApplyLeave = "userApplyLeave"
const val api_userChangeLeaveStatus = "userChangeLeaveStatus"

/****************************** HomePage Basic Version  */
const val api_getBasicVersionHomeDetails = "getBasicVersionHomeDetails"
/************************************************************************/


/*************************** DOT Api  */
const val api_addbubbleRatings = "addbubbleRatings"
const val api_getBubbleRatingList = "getBubbleRatingList"
const val api_getBubbleFromUserRating = "getBubbleFromUserRating"

//not user in app
const val api_getDOTValuesList = "getDOTValuesList"
/************************************************************************/


/***************************** Action APi  */
const val api_add_rating = "ratingsToDotValues"
const val api_actionTierList = "actionTierList"
const val getActionDetail = "getActionDetail"

/************************ Know Org  */
const val getAllOfficenDepartments = "getAllOfficenDepartments"
const val getDepartmentList = "getDepartmentList"
const val getKnowOrganisationDetails = "getKnowOrganisationDetails"

const val api_getUserDashboardReport = "getUserDashboardReport"
const val api_getHappyIndexMonthCount = "getHappyIndexMonthCount"
const val api_getHappyIndexWeeksCount = "getHappyIndexWeeksCount"
const val api_getHappyIndexDaysCount = "getHappyIndexDaysCount"
var getKudosAward = "getKudosAward"
/****************************************************************************/


/*************************************** Know Member  */
const val getUserByType = "getUserByType"
const val getKnowMemberDetails = "getKnowMemberDetails"
/************************************************************************/


/***************************** Offloading  */
const val getFeedbackDetail = "getFeedbackDetail"
const val postFeedback = "postFeedback"
const val getChatMessages = "getChatMessages"
const val iotSendMsg = "iotSendMsg"
const val getInboxChatList = "getInboxChatList"
/*********************************************************************/


/********************************* Notification  */
const val getBubbleRatingUnReadNotificationList = "getBubbleRatingUnReadNotificationList"
const val getBubbleRatingNotificationList = "getBubbleRatingNotificationList"
const val changeNotificationStatus = "changeNotificationStatus"
const val readAllNotification = "readAllNotification"
/**********************************************************************/

/***************************** support  */
const val getSupportHistory = "getSupportHistory"
const val addCustomerSupport = "addCustomerSupport"
const val getUserChatMessages = "getUserChatMessages"
const val sendChatMessage = "sendChatMessage"
/*******************************************************************/


/********************************* UserProfile  */
const val userProfile = "userProfile"
const val userInteractionOnApp = "userInteractionOnApp"
const val userLogout = "userLogout"
const val updateUserProfile = "updateUserProfile"
const val updatePasswordWithCurrentPassword = "updatePasswordWithCurrentPassword"
/*******************************************************************************************/


/********************************* Action  */
const val getActionList = "getActionList"
const val deleteAction = "deleteAction"
const val updateStatus = "updateStatus"
const val getThemeList = "getThemeList"
const val updateAction = "updateAction"
const val addAction = "addAction"
const val addComment = "addComment"
const val listComment = "listComment"
/*********************************************************************************/


/************************************ Org Studies  */
const val getOrgDashboardReportWithFilter = "getOrgDashboardReportWithFilter"

/** Free Version */
const val freeVersionHomeDetails = "getFreeVersionHomeDetails"

/*******************************  Questions and Update Api  */
const val isCOTanswerDone = "isCOTanswerDone"
const val addCOTAnswer = "addCOTAnswer"
const val api_addCOTfunctionalLensAnswer = "addCOTfunctionalLensAnswer"
const val api_sot_addSOTanswers = "addSOTanswers"
const val api_sot_addSOTQusAns = "getSOTquestionAnswers"
const val api_sot_updateSOTquestionAnswer = "updateSOTquestionAnswer"
const val api_addSOTmotivationAnswer = "addSOTmotivationAnswer"

/**                             Questions Api                 */
const val api_sot_quest = "getSotQuestionList"
const val api_getSOTmotivationUserList = "getSOTmotivationUserList"
const val api_getSOTmotivationQuestions = "getSOTmotivationQuestions"
const val api_fubction_qusList = "getCOTfunctionalLensQuestionsList"
const val api_getCOTMapperSummary = "getCOTMapperSummary"
const val api_functionallensdetail = "getCOTFunctionalLensDetail"
const val getCOTindividualSummary = "getCOTindividualSummary"
const val api_get_COTQuestions = "getCOTQuestions"
const val api_getCOTfuncLensCompletedAnswers = "getCOTfuncLensCompletedAnswers"
const val api_getSOTmotivationCompletedAnswer = "getSOTmotivationCompletedAnswer"

/**                            update Api                  */
const val api_updateCOTteamRoleMapAnswers = "updateCOTteamRoleMapAnswers"
const val api_updateCOTfunLensAnswers = "updateCOTfunLensAnswers"
const val api_updateSOTmotivationAnswers = "updateSOTmotivationAnswers"

/**                      get Graph                      */
const val api_getTribeometerSubGraph = "getTribeometerSubGraph"
const val api_getTribeometerMainSubGraph = "getTribeometerMainSubGraph"
/**************************************************************************************/


/*************************************** Personality Type  */
const val api_getPersonalityTypeReport = "getPersonalityTypeReport"
const val api_getPersonalityTypeReportSubGraph = "getPersonalityTypeReportSubGraph"
const val api_getPersonalityTypeReportUserSubGraph = "getPersonalityTypeReportUserSubGraph"
const val api_getPersonalityTypeQuestionList = "getPersonalityTypeQuestionList"
const val api_getPersonalityTypeCompletedAnswers = "getPersonalityTypeCompletedAnswers"
const val api_updatePersonalityTypeAnswers = "updatePersonalityTypeAnswers"
const val api_addPersonalityTypeAnswers = "addPersonalityTypeAnswers"
/***************************************************************************************/


/*********************************************** Risk Module  */
const val api_riskRegisterList = "riskRegisterList"
const val api_riskDetail = "riskDetail"

/**************************************** update Notification push  */
const val api_updatePushNotificationStatus = "updatePushNotificationStatus"
